code/datums/components/edible.dm

```diff
@@ -0,0 +1,271 @@
+/*!
+
+This component makes it possible to make things edible. What this means is that you can take a bite or force someone to take a bite (in the case of items).
+These items take a specific time to eat, and can do most of the things our original food items could.
+
+Behavior that's still missing from this component that original food items had that should either be put into seperate components or somewhere else:
+	Components:
+	Drying component (jerky etc)
+	Customizable component (custom pizzas etc)
+	Processable component (Slicing and cooking behavior essentialy, making it go from item A to B when conditions are met.)
+	Dunkable component (Dunking things into reagent containers to absorb a specific amount of reagents)
+
+	Misc:
+	Something for cakes (You can store things inside)
+
+*/
+/datum/component/edible
+	///Amount of reagents taken per bite
+	var/bite_consumption = 2
+	///Amount of bites taken so far
+	var/bitecount = 0
+	///Flags for food
+	var/food_flags = NONE
+	///Bitfield of the types of this food
+	var/foodtypes = NONE
+	///Size of the food's volume
+	var/volume = 50
+	///Amount of seconds it takes to eat this food
+	var/eat_time = 30
+	///List of tastes players can taste when they consume something, for example list("crisps" = 2, "salt" = 1)
+	var/list/tastes = list("nothing")
+	///Defines how much it lowers someones satiety (Need to eat, essentialy)
+	var/junkiness = 0
+	///Message to send when eating
+	var/list/eatverbs
+	///Callback to be ran for when you take a bite of something
+	var/datum/callback/after_eat
+	///Atom that owns this component
+	var/atom/owner
+	///Last time we checked for food likes
+	var/last_check_time
+
+
+	//var/dried_type = null //move this to obj/item/dry-able or to a dry component
+	//var/dry = 0 //move this to obj/item/dry-able or to a dry component
+
+
+	///Should be done through some kind of seperate cooking code
+	//var/cooked_type = null  //should be on the item itself.
+
+
+	///Custom food component
+	//var/filling_color = "#FFFFFF" //color to use when added to custom food.
+	//var/custom_food_type = null  //for food customizing. path of the custom food to create
+	//var/customfoodfilling = 1 // whether it can be used as filling in custom food
+
+
+	///THIS NEEDS TO BE MOVED TO A PROCCESABLE COMPONENT. Allowing you to use an item to transform an item into another item. E.G. cutting, smashing, etc
+	/// Type path For cut-able food.
+	///var/slice_path = null
+	/// Amount of slices for cutting
+	///var/slices_num = 0
+	/// Time it takes to cut food
+	///var/slice_duration = 20

```

Comment:
Dont keep commented stuff if you're not using it yet

Reply:
I'm scared I'll forget :( what can I do